### PR TITLE
Tiff decoding

### PR DESCRIPTION
**Pull request by szalpal**
Added class for tiff image decoding, on top of last changes in refactoring `image` module

### CODE REVIEW

**Comment by szalpal** [comment]
Tests pass 37987

File: dali/image/tiff.cc [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+      const auto value_count = buffer.Read<uint32_t>(entry_offset + 4);
+      DALI_ENFORCE(value_count == 1);
+
+      size_t value;
+      if (value_type == TYPE_WORD) {
+        value = buffer.Read<uint16_t>(entry_offset + 8);
+      } else if (value_type == TYPE_DWORD) {
+        value = buffer.Read<uint32_t>(entry_offset + 8);
+      } else {
+        DALI_FAIL("Couldn't read Tiff image dims.");
```
Comment (Kh4L) [inline dali/image/tiff.cc]: Small detail, but it should be "Couldn't read TIFF image dims."

File: dali/image/tiff.h [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+  void to_little_endian(T *value) {
+    static_assert(std::is_integral<T>::value, "Converting floating point value unsupported");
+
+    char *value_bytes = reinterpret_cast<char *>(value);
+    std::vector<char> value_copy(value_bytes, value_bytes + sizeof(T));
+
+    {
+      int i = 0;
+      for (auto it = value_copy.rbegin(); it != value_copy.rend(); ++it, ++i) {
+        value_bytes[i] = *it;
```
[the remark this answers is not in the document]
Comment (szalpal) [reply dali/image/tiff.h]: Done

File: dali/image/tiff.cc (hunk 1 above)
Comment (szalpal) [reply dali/image/tiff.cc]: Done

**Pull request merged**